WordPress/Classes/Services/PlanService.swift

```diff
@@ -31,92 +37,87 @@ open class PlanService: LocalCoreDataService {
         let remote = PlanServiceRemote(wordPressComRestApi: api)
         remote.getWpcomPlans({ plans in
 
-            self.mergeRemoteWpcomPlans(plans.plans, remoteGroups: plans.groups, remoteFeatures: plans.features, onComplete: {
-                success()
-            })
+            self.mergeRemoteWpcomPlans(plans.plans, remoteGroups: plans.groups, remoteFeatures: plans.features, onComplete: success)
 
         }, failure: failure)
     }
 
-    func mergeRemoteWpcomPlans(_ remotePlans: [RemoteWpcomPlan],
+    private func mergeRemoteWpcomPlans(_ remotePlans: [RemoteWpcomPlan],
                                remoteGroups: [RemotePlanGroup],
                                remoteFeatures: [RemotePlanFeature],
                                onComplete: @escaping () -> Void ) {
-
-        mergeRemoteWpcomPlans(remotePlans)
-        mergeRemotePlanGroups(remoteGroups)
-        mergeRemotePlanFeatures(remoteFeatures)
-
-        ContextManager.sharedInstance().save(managedObjectContext) {
-            onComplete()
-        }
+        coreDataStack.performAndSave({ context in
+            self.mergeRemoteWpcomPlans(remotePlans, in: context)
+            self.mergeRemotePlanGroups(remoteGroups, in: context)
+            self.mergeRemotePlanFeatures(remoteFeatures, in: context)
+        }, completion: onComplete)
     }
 
 
-    func allPlans() -> [Plan] {
+    func allPlans(in context: NSManagedObjectContext) -> [Plan] {
         let fetchRequest = NSFetchRequest<NSFetchRequestResult>(entityName: "Plan")
         fetchRequest.sortDescriptors = [NSSortDescriptor(key: "order", ascending: true)]
         do {
-            return try managedObjectContext.fetch(fetchRequest) as! [Plan]
+            return try context.fetch(fetchRequest) as! [Plan]
         } catch let error as NSError {
             DDLogError("Error fetching Plans: \(error.localizedDescription)")
             return [Plan]()
         }
     }
 
 
-    func findPlanByShortname(_ shortname: String) -> Plan? {
-        let plans = allPlans() as NSArray
+    private func findPlanByShortname(_ shortname: String, in context: NSManagedObjectContext) -> Plan? {
+        let plans = allPlans(in: context) as NSArray
         let results = plans.filtered(using: NSPredicate(format: "shortname = %@", shortname))
         return results.first as? Plan
```

Comment:
Looking at the code, given `allPlans(in:)` returns `[Plan]`, I think we should be able to remove the double type cast? Or am I missing something?

```suggestion
        return allPlans(in: context).filter { $0.shortname == shortname }.first
```

Reply:
Good catch! Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/19949/commits/bf2bf5b11afed12b484d3df3b67dfa43941e2f81